python/ray/tests/test_memory_deadlock.py

```diff
@@ -68,35 +63,46 @@ def nested_tasks_parent():
 def test_deadlock_task_with_nested_task(
     ray_with_memory_monitor,
 ):
-    with pytest.raises(ray.exceptions.OutOfMemoryError) as _:
-        ray.get(nested_tasks_parent.remote())
+    with pytest.raises(
+        (ray.exceptions.RayTaskError, ray.exceptions.OutOfMemoryError)
+    ) as _:
+        bytes1 = get_additional_bytes_to_reach_memory_usage_pct(
+            memory_usage_threshold_fraction - 0.1
+        )
+        bytes2 = (
+            get_additional_bytes_to_reach_memory_usage_pct(
+                memory_usage_threshold_fraction + 0.2
+            )
+            - bytes1
+        )
+        ray.get(task_with_nested_task.remote(bytes1, bytes2, None, None))
 
 
 @ray.remote
-def nested_actor_and_task_parent():
+def task_with_nested_actor(fraction1, fraction2):
     leaker = Leaker.options(max_restarts=1, max_task_retries=1).remote()
-    actor_bytes = get_additional_bytes_to_reach_memory_usage_pct(0.7)
+    actor_bytes = get_additional_bytes_to_reach_memory_usage_pct(fraction1)
     ray.get(leaker.allocate.remote(actor_bytes))
 
-    parent_bytes = get_additional_bytes_to_reach_memory_usage_pct(0.9)
+    parent_bytes = get_additional_bytes_to_reach_memory_usage_pct(fraction2)
     dummy = alloc_mem(parent_bytes)
```

Comment:
can we add a arg to test both variants where it complete the nested actor then allocate in this task, and vice versa